src/entity/mesh.hpp

```diff
@@ -186,72 +190,73 @@ generate_mesh(T voxel_object)
-                                voxel_position[1] + off_set_1[1] + off_set_2[1],
-                                voxel_position[2] + off_set_1[2] + off_set_2[2]
+                            glm::ivec3(
+                                voxel_position[0] + offset_1[0] + offset_2[0],
+                                voxel_position[1] + offset_1[1] + offset_2[1],
+                                voxel_position[2] + offset_1[2] + offset_2[2]
                             )
-                            + off_set
+                            + offset
                         );
                         indexed_vertices.push_back(
-                            glm::vec3(
-                                voxel_position[0] + off_set_2[0],
-                                voxel_position[1] + off_set_2[1],
-                                voxel_position[2] + off_set_2[2]
+                            glm::ivec3(
+                                voxel_position[0] + offset_2[0],
+                                voxel_position[1] + offset_2[1],
+                                voxel_position[2] + offset_2[2]
                             )
-                            + off_set
+                            + offset
                         );
 
-                        uint32_t int_color = color.second;
-                        uint32_t red = (int_color >> 24) & 0xFF;
-                        uint32_t green = (int_color >> 16) & 0xFF;
-                        uint32_t blue = (int_color >> 8) & 0xFF;
+                        //uint32_t int_color = color.second;
+                        //uint32_t red = (int_color >> 24) & 0xFF;
+                        //uint32_t green = (int_color >> 16) & 0xFF;
+                        //uint32_t blue = (int_color >> 8) & 0xFF;
                         // the last one >> 0 is A
-                        glm::vec3 vector_color(
-                            red / 255.0, green / 255.0, blue / 255.0
-                        );
+                        //glm::vec3 vector_color(
+                        //    red / 255.0, green / 255.0, blue / 255.0
+                        //);
                         // how many corners on a square are there?
                         for (size_t voxel_position = 0; voxel_position < 4;
                              voxel_position++) {
-                            indexed_colors.push_back(vector_color);
+                            indexed_colors.push_back(color.second);
                         }
 
-                        glm::vec3 triangle_normal = glm::normalize(glm::cross(
-                            indexed_vertices[vertex_size]
-                                - indexed_vertices[vertex_size + 1],
-                            indexed_vertices[vertex_size]
-                                - indexed_vertices[vertex_size + 2]
-                        ));
+                        glm::i8vec3 triangle_normal =
+                            glm::ivec3(glm::normalize(glm::cross(glm::vec3(
```

Comment:
You're doing an implicit narrowing conversion

Reply:
I guess I could make it explicit.
I realize there might be a bug here, as normalize might return (0.0001, .000000001, .9999999) -> (0,0,0)